src/search_results_handler/aggregator.rs

```diff
@@ -58,8 +58,19 @@ pub async fn aggregate(
         searx::results(query, page, &user_agent)
     );
 
-    let ddg_map_results: HashMap<String, RawSearchResult> = ddg_map_results?;
-    let searx_map_results: HashMap<String, RawSearchResult> = searx_map_results?;
+    let ddg_map_results = ddg_map_results.unwrap_or_else(|e| {
+        if debug {
+            log::error!("Error fetching results from DuckDuckGo: {:?}", e);
+        }
+        HashMap::new()
+    });
+
+    let searx_map_results = searx_map_results.unwrap_or_else(|e| {
+        if debug {
+            log::error!("Error fetching results from Searx: {:?}", e);
+        }
+        HashMap::new()
+    });
```

Comment:
> Also, after testing your changes it seems that the errors are not getting logged on stdout when they occur so I would suggest making the following changes in order to get this fixed

I have logged the errors here

Reply:
Ok, I think this is perfect :+1: . Thanks :heart: for working on this and putting in the effort :slightly_smiling_face: .